lib/functionsConfig.js

```diff
@@ -60,22 +60,24 @@ exports.getFirebaseConfig = function(projectId, instance) {
 };
 
 
+/* If you make changes to this function, run "node scripts/test-functions-config.js"
+   to ensure that nothing broke. */
 exports.setVariablesRecursive = function(projectId, configId, varPath, val) {
   try {
     var parsed = JSON.parse(val);
-    if (!_.isPlainObject(parsed)) { // could be a number, etc.
-      return _setVariable(projectId, configId, varPath, parsed);
+    if (!_.isPlainObject(parsed)) {
+      // parsed is a number
```

Comment:
Couldn't this also have been a quoted string? What happens if I do:

    firebase functions:config:set foo.bar='"baz"'